lib/portlayer/attach/communication/connector.go

```diff
@@ -112,33 +128,46 @@ func (c *Connector) Interaction(ctx context.Context, id string) (SessionInteract
 func (c *Connector) interaction(ctx context.Context, id string) (SessionInteractor, error) {
 	defer trace.End(trace.Begin(id))
 
-	conn := c.aliveAndKicking(ctx, id)
-	if conn != nil {
-		return conn, nil
-	}
-
 	if ctx.Err() == context.DeadlineExceeded {
 		return nil, fmt.Errorf("attach connector: no such connection")
 	}
 
+	conn, err := c.SessionIfAlive(ctx, id)
+	if conn != nil && err == nil {
+		return conn, nil
+	}
+
 	result := make(chan SessionInteractor, 1)
 	go func() {
 		ok := false
-		var conn SessionInteractor
+		var v *LazySessionInteractor
 
 		c.mutex.RLock()
 		defer c.mutex.RUnlock()
 
 		for !ok && ctx.Err() == nil {
-			conn, ok = c.interactions[id]
+			v, ok = c.interactions[id]
 			if ok {
-				// no need to test this as we just created it
-				if err := conn.Unblock(); err == nil {
-					log.Infof("attach connector: Unblocked %s, returning", id)
+				conn, err := v.Initialize()
+				if err != nil {
+					log.Error(err)
+					ok = false
```

Comment:
looks like it will break for loop for err happens, so result channel is not set, and context is not done yet, but the select https://github.com/vmware/vic/pull/5300/files#diff-4a8e153ac25578d27311e828f2772579R178 here does not handle that situation.